components/globals/Alert/Alert.tsx

```diff
@@ -0,0 +1,310 @@
+import { CircleCheckIcon, InfoIcon, WarningIcon } from "@components/form-builder/icons";
+import React, { ReactNode, useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+export enum ErrorStatus {
+  SUCCESS = "SUCCESS",
+  WARNING = "WARNING",
+  ERROR = "ERROR",
+  INFO = "INFO",
+}
+
+export const Title = ({
+  children,
+  level = "h2",
+  status,
+}: {
+  children: string;
+  level?: "h2" | "h3" | "h4";
+  status?: ErrorStatus;
+}) => {
+  if (typeof children !== "string") {
+    return <>{children}</>;
+  }
+  const Wrapper = `${level}` as keyof JSX.IntrinsicElements;
+  const className = status ? `${defaultClasses.text[status]}` : "";
+
+  return <Wrapper className={`${className}`}>{children}</Wrapper>;
+};
+
+export const Body = ({ children }: { children: JSX.Element | string }) => {
+  return <>{children}</>;
+};
+
+export const Icon = ({
+  children,
+  status,
+}: {
+  children: JSX.Element;
+  status?: ErrorStatus | undefined;
+}) => {
+  if (children.type.name === "Icon") {
+    return <>{children}</>;
+  }
+  const className = status ? `${defaultClasses.icon[status]}` : "";
+
+  return <div className={`mr-1 ${className}`}>{children}</div>;
+};
+
+const defaultClasses = {
+  background: {
+    [ErrorStatus.SUCCESS]: "bg-emerald-50",
+    [ErrorStatus.WARNING]: "bg-yellow-50",
+    [ErrorStatus.ERROR]: "bg-red-50",
+    [ErrorStatus.INFO]: "bg-indigo-50",
+  },
+  icon: {
+    [ErrorStatus.SUCCESS]: "[&_svg]:fill-emerald-700",
+    [ErrorStatus.WARNING]: "[&_svg]:fill-slate-950",
+    [ErrorStatus.ERROR]: "[&_svg]:fill-red-700",
+    [ErrorStatus.INFO]: "[&_svg]:fill-slate-950",
+  },
+  text: {
+    [ErrorStatus.SUCCESS]: "text-emerald-700",
+    [ErrorStatus.WARNING]: "text-yellow-700",
+    [ErrorStatus.ERROR]: "text-red-700",
+    [ErrorStatus.INFO]: "text-indigo-700",
+  },
+};
+
+const defaultIcons = {
+  [ErrorStatus.SUCCESS]: (
+    <Icon status={ErrorStatus.SUCCESS}>
+      <CircleCheckIcon className={`h-12 w-12`} />
```

Comment:
wondering if we should add a const or something for the className so each case can just use vs "className={`h-12 w-12`} />" on each